docs/03_server/04_event-handler/03_advanced.md

```diff
@@ -241,110 +251,247 @@ eventHandler {
 
-The `APPROVAL` record is keyed on an auto-generated `APPROVAL_ID` and does not have a direct link to the original record. You have to create a link by adding “approval entity” details to the payload returned on an event ack in the `onValidate` method. These details include the `ENTITY_TABLE` and `ENTITY_KEY`. This allows you to decide how to identify the original record (e.g. creating a compound key in the case of multi-field keys). When the approval entity details are provided, the platform creates a record in the `APPROVAL_ENTITY` table and populates it with the details provided, and the `APPROVAL_ID` of the `APPROVAL` record. There is also an `APPROVAL_ENTITY_COUNTER`, which is populated by the AUTH_CONSOLIDATOR process by default; this can be handy in order to easily know how many approvals are pending for a given entity.
+There are other useful properties we can set as part of the `approvableAck` definition. All of them are optional and the whole list is detailed below:
+* `entityDetails` is a list of `ApprovalEntityDetails` with their corresponding `entityTable`, `entityId` and `approvalType` properties (see previous paragraph). By default this list is empty.
+* `approvalMessage` contains the text that will be sent back to the client assuming the event is successfully submitted for approval. By default, we return "Your request was successful and has been submitted for approval".
+* `additionalDetails` can provide context information that is only available from a server-side perspective. This information complements the `APPROVAL_MESSAGE` content provided by the frontend.
+* `approvalType` functions the same way as it does within `entityDetails`. However, `entityDetails` requires an `entityId` value, which means there is an assumption that we are attempting to amend or delete an existing database record. If this is a new record insertion, you can set the `approvalType` at this level to provide additional information about what this pending approval is meant to do. Valid values are `NEW`, `UPDATE`, `REMOVE` and `UNKNOWN` (which is also the default value).
```

Comment:
This should start with more of a description. "approvalType is used to state the action that will happen upon this event being approved, NEW for insertions, UPDATE for amends, REMOVE for removals. Most events will be simple but of course some may affect multiple entities in different ways, which is why entityDetails can contain many entities each with their own approvalType"

I feel like this might need a bit of an example later on too (not all in one or it could get confusing)/seem complex

Reply:
I will add the extra info and also a comment explaining you can have multiple approval entities as part of the code example.